packages/flutter/lib/src/rendering/table.dart

```diff
@@ -0,0 +1,784 @@
+    BorderSide outside: BorderSide.none
+  }) {
+    return new TableBorder(
+      top: outside,
+      right: outside,
+      bottom: outside,
+      left: outside,
+      horizontalInside: inside,
+      verticalInside: inside
+    );
+  }
+
+  final BorderSide horizontalInside;
+
+  final BorderSide verticalInside;
+
+  @override
+  TableBorder scale(double t) {
+    return new TableBorder(
+      top: top.copyWith(width: t * top.width),
+      right: right.copyWith(width: t * right.width),
+      bottom: bottom.copyWith(width: t * bottom.width),
+      left: left.copyWith(width: t * left.width),
+      horizontalInside: horizontalInside.copyWith(width: t * horizontalInside.width),
+      verticalInside: verticalInside.copyWith(width: t * verticalInside.width)
+    );
+  }
+
+  static TableBorder lerp(TableBorder a, TableBorder b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return b.scale(t);
+    if (b == null)
+      return a.scale(1.0 - t);
+    return new TableBorder(
+      top: BorderSide.lerp(a.top, b.top, t),
+      right: BorderSide.lerp(a.right, b.right, t),
+      bottom: BorderSide.lerp(a.bottom, b.bottom, t),
+      left: BorderSide.lerp(a.left, b.left, t),
+      horizontalInside: BorderSide.lerp(a.horizontalInside, b.horizontalInside, t),
+      verticalInside: BorderSide.lerp(a.verticalInside, b.verticalInside, t)
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (super != other)
+      return false;
+    final TableBorder typedOther = other;
+    return horizontalInside == typedOther.horizontalInside &&
+           verticalInside == typedOther.verticalInside;
+  }
+
+  @override
+  int get hashCode => hashValues(super.hashCode, horizontalInside, verticalInside);
+
+  @override
+  String toString() => 'TableBorder($top, $right, $bottom, $left, $horizontalInside, $verticalInside)';
+}
+
+/// Vertical alignment options for cells in [RenderTable] objects.
+///
+/// This is specified using [TableCellParentData] objects on the
+/// [RenderObject.parentData] of the children of the [RenderTable].
+enum CellVerticalAlignment {
+  /// Cells with this alignment are placed with their top at the top of the row.
+  top,
+
+  /// Cells with this alignment are vertically centered in the row.
+  middle,
+
+  /// Cells with this alignment are placed with their bottom at the bottom of the row.
+  bottom,
+
+  /// Cells with this alignment are aligned such that they all share the same
+  /// baseline. Cells with no baseline are top-aligned instead. The baseline
+  /// used is specified by [RenderTable.baseline]. It is not valid to use the
+  /// baseline value if [RenderTable.baseline] is not specified.
+  baseline,
+
+  /// Cells with this alignment are sized to be as tall as the row, then made to fit the row.
+  /// If all the cells have this alignment, then the row will have zero height.
+  fill
+}
+
+class RenderTable extends RenderBox {
```

Comment:
yeah, this needs tons of docs. I'm not doing it in this PR because the API might change and I need a fresh perspective before I write the docs.